File: built-in plugins dsp/mb-mic-plosive.cpp
/**
 * MB Plosive Reducer
 * Category : effect
 * Type     : microphone
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Pops and plosive control
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_MIC_PLOSIVE_H
#define MB_MIC_PLOSIVE_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbMicPlosive : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-mic-plosive";
    static constexpr const char* PLUGIN_NAME    = "MB Plosive Reducer";
    static constexpr const char* PLUGIN_TYPE    = "microphone";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float sensitivity = 0.5f;  // range [0, 1]
    float freq = 120f;  // range [50, 300]
    float reduction = 12f;  // range [0, 24]
    };

    MbMicPlosive() = default;
    ~MbMicPlosive() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.sensitivity = std::clamp(params.sensitivity, 0f, 1f);
        params.freq = std::clamp(params.freq, 50f, 300f);
        params.reduction = std::clamp(params.reduction, 0f, 24f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Plosive Reducer
        return input;
    }
};

#endif // MB_MIC_PLOSIVE_H

File: built-in plugins dsp/mb-mic-plosive.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbMicPlosivePlugin: PluginDefinition = { id: 'mb-mic-plosive', slug: 'mb-mic-plosive', name: 'MB Plosive Reducer', category: 'effect', type: 'microphone', version: '1.0.0', description: 'Pops and plosive control', author: 'Max Booster', grade: 'A', parameters: [{ id: 'sensitivity', name: 'Sensitivity', type: 'float', defaultValue: 0.5, minValue: 0, maxValue: 1, automatable: true }, { id: 'freq', name: 'Frequency', type: 'float', defaultValue: 120, minValue: 50, maxValue: 300, automatable: true }, { id: 'reduction', name: 'Reduction', type: 'float', defaultValue: 12, minValue: 0, maxValue: 24, automatable: true }], defaultPreset: { sensitivity: 0.5, freq: 120, reduction: 12 } };

export default MbMicPlosivePlugin;
